content/docs/reference/cli-branches.md

```diff
@@ -0,0 +1,198 @@
+│ Id                      │ Name                    │ Created At           │
+├─────────────────────────┼─────────────────────────┼──────────────────────┤
+│ br-withered-king-763176 │ br-withered-king-763176 │ 2023-06-19T22:35:25Z │
+└─────────────────────────┴─────────────────────────┴──────────────────────┘
+```
+
+### update
+
+This subcommand allows you to update a branch in a Neon project.
+
+#### Usage
+
+```bash
+neonctl branches update [options]
+```
+
+#### Options
+
+In addition to the Neon CLI [global options](../neon-cli/global-options), the `update` subcommand supports these options:
+
+| Option        | Description | Type   | Required  |
+| ------------- | ----------- | ------ | :-----: |
+| --project.id  | Project ID  | string | &check; |
+| --branch.name |             | string |         |
+| --branch.id   | Branch ID   | string | &check; |
+
+#### Example
+
+```bash
+neonctl branches update --project.id spring-sky-578180 --branch.id br-withered-king-763176 --branch.name mynewbranch
+┌─────────────────────────┬─────────────┬──────────────────────┐
+│ Id                      │ Name        │ Created At           │
+├─────────────────────────┼─────────────┼──────────────────────┤
+│ br-withered-king-763176 │ mynewbranch │ 2023-06-19T22:35:25Z │
+└─────────────────────────┴─────────────┴──────────────────────┘
+```
+
+### delete
+
+This subcommand allows you to delete a branch in a Neon project.
+
+#### Usage
+
+```bash
+neonctl branches delete [options]
+```
+
+#### Options
+
+In addition to the Neon CLI [global options](../neon-cli/global-options), the `delete` subcommand supports these options:
+
+| Option        | Description | Type   | Required  |
+| ------------- | ----------- | ------ | :------: |
+| --project.id  | Project ID  | string | &check; |
+| --branch.id   | Branch ID   | string | &check; |
+
+#### Example
+
+<CodeBlock shouldWrap>
+
+```bash
+neonctl branches delete --project.id spring-sky-578180 --branch.id br-withered-king-763176
+┌─────────────────────────┬─────────────┬──────────────────────┐
+│ Id                      │ Name        │ Created At           │
+├─────────────────────────┼─────────────┼──────────────────────┤
+│ br-withered-king-763176 │ mynewbranch │ 2023-06-19T22:35:25Z │
+└─────────────────────────┴─────────────┴──────────────────────┘
+```
+
+</CodeBlock>
+
+### get
+
+This subcommand allows you to retrieve details about a branch.
+
+#### Usage
+
+```bash
+neonctl branches get [options]
+```
+
+#### Options
+
+In addition to the Neon CLI [global options](../neon-cli/global-options), the `get` subcommand supports this option:
+
+#### Options
+
+| Option        | Description | Type   | Required |
+| ------------- | ----------- | ------ | :------: |
+| --project.id  | Project ID  | string | &check; |
+| --branch.id   | Branch ID   | string | &check; |
+
+#### Example
+
+<CodeBlock shouldWrap>
+
+```bash
+neonctl branches get --project.id spring-sky-578180 --branch.id br-sweet-sun-522796 
```

Comment:
@duskpoet , should we also return more info about a branch?
the usage data, the list of endpoints and types
similar to what we do on the branch page in the UI

Reply:
not sure how to fit all that, maybe with @tombrewsviews help :)